exchanges/coinbasepro/ratelimit.go

```diff
@@ -4,35 +4,56 @@ import (
 	"context"
 	"time"
 
+	"github.com/pkg/errors"
 	"github.com/thrasher-corp/gocryptotrader/exchanges/request"
 	"golang.org/x/time/rate"
 )
 
-// Coinbasepro rate limit conts
+// Coinbasepro rate limit constants
 const (
-	coinbaseproRateInterval = time.Second
-	coinbaseproAuthRate     = 5
-	coinbaseproUnauthRate   = 2
+	coinbaseV3Interval = time.Second
+	coinbaseV3Rate     = 27
+
+	coinbaseV2Interval = time.Hour
+	coinbaseV2Rate     = 10000
+
+	coinbaseWSInterval = time.Second
+	coinbaseWSRate     = 750
+)
+
+// Coinbase pro rate limits
+const (
+	V2Rate request.EndpointLimit = iota
+	V3Rate
+	WSRate
 )
 
 // RateLimit implements the request.Limiter interface
 type RateLimit struct {
-	Auth   *rate.Limiter
-	UnAuth *rate.Limiter
+	RateLimV3 *rate.Limiter
+	RateLimV2 *rate.Limiter
+	RateLimWS *rate.Limiter
 }
 
 // Limit limits outbound calls
 func (r *RateLimit) Limit(ctx context.Context, f request.EndpointLimit) error {
-	if f == request.Auth {
-		return r.Auth.Wait(ctx)
+	switch f {
+	case V3Rate:
+		return r.RateLimV3.Wait(ctx)
+	case V2Rate:
+		return r.RateLimV2.Wait(ctx)
+	case WSRate:
+		return r.RateLimWS.Wait(ctx)
+	default:
+		return errors.Errorf(errUnknownEndpointLimit, f)
```

Comment:
We typically format errors like the following:
```suggestion
		return fmt.Errorf("%w %v",errUnknownEndpointLimit, f)
```
Then, you should be changing `errUnknownSide` to the following:
```go
var (
	errUnknownSide = errors.New("unknown side")
)
```

That way, in test `TestProcessSnapshotUpdate` you can do the following check:
`assert.ErrorIs(t, err, errUnknownSide)`

Or, as a complete example:
```go
func TestProcessSnapshotUpdate(t *testing.T) {
	t.Parallel()
	req := WebsocketOrderbookDataHolder{Changes: []WebsocketOrderbookData{{Side: "fakeside", PriceLevel: 1.1,
		NewQuantity: 2.2}}, ProductID: currency.NewBTCUSD()}
	err := c.ProcessSnapshot(&req, time.Time{})
	assert.ErrorIs(t, err, errUnknownSide)

	req.Changes[0].Side = "offer"
	err = c.ProcessSnapshot(&req, time.Now())
	assert.NoError(t, err)

	err = c.ProcessUpdate(&req, time.Now())
	assert.NoError(t, err)
}
```
`fmt.Errorf` allows for rich details to be added to errors, which is helpful for the user. In addition, golang allows you to wrap multiple errors together. Wrapping multiple errors allow for good debugging techniques where we can handle scenarios like

```go
if err != nil {
    if  errors.Is(err, errTheThingBlewUp) {
         // do soemthing specific to address the error
    } else {
        return nil, err
    }
}
```

Reply:
So what I'd like you to do is update all the const `err` you have declared in the `coinbasepro` package, and change them to `errors.New("...")` type errors. Then you can change the error handling in your tests to better parse what's going on, instead of relying on string comparisons. Make them like my example above

Reply:
I've changed most of these, but left errIntervalNotSupported (never used sprintf formatting; was used to maintain every return value being a string) and warnSequenceIssue (kept as a string to distinguish this recoverable issue of data loss from a more pressing error, since only string-based functions are used on it, and since strings passed to websocketDataHandler are logged differently from errors).

If you want me to change those too, let me know.